wrappers/kn/src/nativeMain/kotlin/zxingcpp/Barcode.kt

```diff
@@ -0,0 +1,188 @@
+		if (text != other.text) return false
+		if (ecLevel != other.ecLevel) return false
+		if (symbologyIdentifier != other.symbologyIdentifier) return false
+		if (position != other.position) return false
+		if (orientation != other.orientation) return false
+		if (hasECI != other.hasECI) return false
+		if (isInverted != other.isInverted) return false
+		if (isMirrored != other.isMirrored) return false
+		if (lineCount != other.lineCount) return false
+
+		return true
+	}
+
+	override fun hashCode(): Int {
+		var result = isValid.hashCode()
+		result = 31 * result + (errorMsg?.hashCode() ?: 0)
+		result = 31 * result + format.hashCode()
+		result = 31 * result + contentType.hashCode()
+		result = 31 * result + (text?.hashCode() ?: 0)
+		result = 31 * result + (ecLevel?.hashCode() ?: 0)
+		result = 31 * result + (symbologyIdentifier?.hashCode() ?: 0)
+		result = 31 * result + position.hashCode()
+		result = 31 * result + orientation.hashCode()
+		result = 31 * result + hasECI.hashCode()
+		result = 31 * result + isInverted.hashCode()
+		result = 31 * result + isMirrored.hashCode()
+		result = 31 * result + lineCount.hashCode()
+		return result
+	}
+
+	override fun toString(): String {
+		return "Barcode(" +
+			"isValid=$isValid, " +
+			"errorMsg=$errorMsg, " +
+			"format=$format, " +
+			"contentType=$contentType, " +
+			"text=$text, " +
+			"ecLevel=$ecLevel, " +
+			"symbologyIdentifier=$symbologyIdentifier, " +
+			"position=$position, " +
+			"orientation=$orientation, " +
+			"hasECI=$hasECI, " +
+			"isInverted=$isInverted, " +
+			"isMirrored=$isMirrored, " +
+			"lineCount=$lineCount" +
+			")"
+	}
+}
+
+class BarcodeImpl(
+	override val isValid: Boolean,
+	override val errorMsg: String?,
+	override val format: BarcodeFormat,
+	override val contentType: ContentType,
+	override val text: String?,
+	override val ecLevel: String?,
+	override val symbologyIdentifier: String?,
+	override val position: Position,
+	override val orientation: Int,
+	override val hasECI: Boolean,
+	override val isInverted: Boolean,
+	override val isMirrored: Boolean,
+	override val lineCount: Int,
+	private val bytes: ByteArray? = null,
+	private val bytesECI: ByteArray? = null,
+) : Barcode() {
+	override fun bytes(len: Int): ByteArray? = bytes?.sliceArray(0 until len)
+	override fun bytesECI(len: Int): ByteArray? = bytesECI?.sliceArray(0 until len)
+}
+
+@OptIn(ExperimentalForeignApi::class)
+class BoundBarcode(val cValue: CValuesRef<zxing_Barcode>) : Barcode() {
+	override val isValid: Boolean
+		get() = zxing_Barcode_isValid(cValue)
+	override val errorMsg: String?
+		get() = zxing_Barcode_errorMsg(cValue)?.toKString()
+	override val format: BarcodeFormat
+		get() = zxing_Barcode_format(cValue).parseIntoBarcodeFormat().first { it != BarcodeFormat.None }
+	override val contentType: ContentType
+		get() = zxing_Barcode_contentType(cValue).toKObject()
+
+	override fun bytes(len: Int): ByteArray? = zxing_Barcode_bytes(cValue, cValuesOf(len))?.readBytes(len)
```

Comment:
As far as I understand, the pointer returned by `zxing_Barcode_bytes()` is never freed? See `zxing_free`. The same goes for all the `char *` functions.

Reply:
Partially fixed in: https://github.com/zxing-cpp/zxing-cpp/pull/719/commits/06495f55786a0f1237c43451a812766bdc31cdb6

Reply:
fixed in: https://github.com/zxing-cpp/zxing-cpp/pull/719/commits/2c669e4945d2d38806d9bffbae6946b6cb805077

Reply:
fixed in: https://github.com/zxing-cpp/zxing-cpp/pull/719/commits/a2b5185fb1e6bade5a917014b835952256e3b499